src/modding/NewModGUI.cs

```diff
@@ -308,4 +385,54 @@ private void ClearError()
     {
         errorDisplay.Text = string.Empty;
     }
+
+    private void ChooseIconButtonPressed()
+    {
+        iconFileDialog.PopupCentered();
+    }
+
+    private void IconFileDialogFileSelected(string path)
+    {
+        if (Path.GetFileName(path) == iconFileDialog.CurrentFile)
+        {
+            iconFile.Text = iconFileDialog.CurrentFile;
+        }
+    }
+
+    private void ChoosePckButtonPressed()
+    {
+        pckFileDialog.PopupCentered();
+    }
+
+    private void PckFileDialogFileSelected(string path)
```

Comment:
aren't it the same as IconFileDialogFileSelected? just with the object itself being different which can just be parameter?

Reply:
Yeah, I guess but I don't think there's a way for signals to have object parameters in the scene editor without doing some weird stuff.

Reply:
The way that is handled elsewhere is to use a string identifier or perhaps a method variant to handle providing extra information from scene file defined signal connections.

Reply:
do you have any examples in the codebase I can see, because I am not entirely sure what you mean...

Reply:
For example the options menu and editor tab buttons just pass a string as a parameter set in the scene, and then the C# code parses that string into an enum value to do the actual work it needs to.